File: RemoveDupes/removedupes.cpp
#include <stdlib.h>
#include <stdio.h>
#include <vector>
#include <iostream>

using namespace std;

class Solution 
{
public:
    int removeDuplicates(vector<int>& nums) 
    {
        //case of nums being an empty array
        if(nums.size() == 0)
        {
            return 0;
        }

        //any other case
        else
        {
            //using two pointers, i, j, to iterate over array
            int i = 0;

            for(int j = 1; j < nums.size(); j++)
            {
                //when we reach a new number
                if(nums[j] != nums[i])
                {
                    //increment i
                    i++;
                    //then set the i to point to the new number
                    nums[i] = nums[j];
                }
            }
            //returning i + 1 because i does not count the unique value at position 0 of array
            return i + 1;
        }
    }
};

File: RemoveDupes/removedupes.java
package RemoveDupes;

public class removedupes {
    class Solution {
        public int removeDuplicates(int[] nums) 
        {

            //case of nums being an empty array
            if(nums.length == 0)
            {
                return 0;
            }

            //if there are values in nums
            else
            {
                //creating pointer i
                int i = 0;

                //creating pointer j
                for(int j = 1; j < nums.length; j++)
                {
                    //j goes thru the whole array
                    if(nums[j] != nums[i])
                    {
                        i++;
                        //i only updates when a new value is pointed at
                        nums[i] = nums[j];
                    }
                }
                //unique value at nums[0] not counted so we add one
                return i + 1;
            }
        }
    }
}
